### ISSUE TITLE
[6.0.0-alpha.1] require-atomic-updates false positive with if statement

### ISSUE THREAD

**Issue by teppeis** [eslint/eslint#11740]
**Tell us about your environment**

* **ESLint Version:** 6.0.0-alpha-1
* **Node Version:** 12.2.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "root": true,
  "parserOptions": {
    "ecmaVersion": 2017
  },
  "rules": {
    "require-atomic-updates": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let flag = false;
async function f() {
  if (!flag) {
    await g();
    flag = true;
  }
}
```

```bash
$ ./node_modules/.bin/eslint index.js
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
  47:5  error  Possible race condition: `flag` might be reassigned based on an outdated value of `flag`  require-atomic-updates
```

**Are you willing to submit a pull request to fix this bug?**

No

----

No problem in ESLint v5:

https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgcmVxdWlyZS1hdG9taWMtdXBkYXRlczogZXJyb3IgKi9cclxuXHJcbmxldCBmbGFnID0gZmFsc2U7XHJcbmFzeW5jIGZ1bmN0aW9uIGYoKSB7XHJcbiAgaWYgKCFmbGFnKSB7XHJcbiAgICBhd2FpdCBnKCk7XHJcbiAgICBmbGFnID0gdHJ1ZTtcclxuICB9XHJcbn1cclxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19

**Comment by mysticatea** [comment]
Thank you for your report.

But I think that this is expected behavior.

1. Get `flag`
1. Do asynchronous operation that depends on the `flag` value.
1. Update `flag`

Therefore, it can cause a race condition. If you call `f` while `g` is running, the second `f` call calls `g` again.

I guess that what you want is below:

```js
let initPromice;
async function f() {
  await (initPromice || (initPromice = g()));
  // do something
}
```

**Comment by teppeis** [comment]
That's right! Thank you.
